src/entrypoint.cpp

```diff
@@ -167,6 +189,23 @@ namespace aperf {
       })
       ->option_text(" ");
 
+    struct SubcommandData {
+    std::string metric_command;
+    std::string metric_name;
+    unsigned int frequency;
+    std::string regular_expression;
+    };
+
+    std::vector<std::string> metric_strs;
+    std::regex pattern_metric("");
+    app.add_option("-m,--metric", metric_strs, "Extra metric to sample "
+                   "based on an external profiler/metric command. Usage: -m "
+                   "c:\"METRIC_COMAND\",n:\"METRIC_NAME\",f:SAMPLELING_FREQUENCY"
+                    "(must be a number), r:\"REGULAR_EXPRESSION\""
```

Comment:
I would mention:

* that the sampling frequency and regular expression are optional.
* what the purpose is for the regular expression.
* that the sampling frequency is in times per second and its default value is 10.

Also, you have a typo in the sampling frequency ;)

Reply:
I added a more complete description, but I don't know if it is displaying nicely in the help command. It might be a bit hard to read the way I wrote it.